src/blocks/remote-data-container/components/modals/DataViewsModal.tsx

```diff
@@ -0,0 +1,57 @@
+import { __ } from '@wordpress/i18n';
+
+import { ModalWithButtonTrigger } from './BaseModal';
+import { useModalState } from '../../hooks/useModalState';
+import { ItemList } from '../item-list/ItemList';
+import { useSearchResults } from '@/blocks/remote-data-container/hooks/useSearchResults';
+import { sendTracksEvent } from '@/blocks/remote-data-container/utils/tracks';
+import { getBlockDataSourceType } from '@/utils/localized-block-data';
+
+interface DataViewsModalProps {
+	blockName: string;
+	headerImage?: string;
+	onSelect: ( data: RemoteDataQueryInput ) => void;
+	queryKey: string;
+	title: string;
+}
+
+export function DataViewsModal( props: DataViewsModalProps ) {
```

Comment:
I personally prefer this style just because the `React.FC` type enforces the return value as well. This is super minor though and we don't have a defined standard.

```suggestion
export const DataViewsModal: React.FC< DataViewsModalProps > = props => {
```